### PR TITLE
Add some documentation on testing interacting features

### PR DESCRIPTION
**Pull request by clelland**
There are several support libraries provided in different directories across WPT which are useful for testing interactions between features, but only if test authors are aware of them. This adds a page under writing-tests where I have started to collect those libraries.

### CODE REVIEW

File: docs/writing-tests/general-guidelines.md [hunk 1]
```diff
@@ -115,7 +115,9 @@ no [parse errors](https://validator.nu).
 
 This is not, however, to discourage testing of edge cases or
 interactions between multiple features; such tests are an essential
-part of ensuring interoperability of the web platform.
+part of ensuring interoperability of the web platform. When possible, use the
+canonical support libraries provided by features; see [here][interacting-features]
+for more information.
```
Comment (sideshowbarker) [inline docs/writing-tests/general-guidelines.md:120]:
```suggestion
canonical support libraries provided by features; for more information, see the documentation on [testing interactions among features][interacting-features].
```

File: docs/writing-tests/interacting-features.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Testing interactions between features
+
+Web platform features do not exist in isolation. Often testing the interaction between two features is necessary in tests.
+To support this, many directories contain libraries which are intended to be used from other directories.
+
+These are not WPT server features, but are canonical usage of one feature intended for other features to test against.
+This allows the tests for a feature to be decoupled as much as possible from the specifics of another feature which it should integrate with.
```
Comment (sideshowbarker) [inline docs/writing-tests/interacting-features.md:7]:
```suggestion
Web platform features do not exist in isolation. Often, testing the interaction between two features is necessary in tests.
To support this, many directories contain libraries which are intended to be used from other directories.

These are not WPT server features, but are canonical usage of one feature intended for other features to test against.
This allows the tests for a feature to be decoupled as much as possible from the specifics of another feature which it should integrate with.
```

File: docs/writing-tests/interacting-features.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Features which need to test their interaction with cookies can use the scripts in cookies/resources to control which cookies are set on a given request.
+
+### Permissions Policy
+
+Features which integrate with Permissions Policy can make use of the permissions-policy.js support library to generate a set of tests for that integration.
+
+### Reporting
+
+Testing integration with the Reporting API can be done with the help of the common report collector. This service will collect reports sent from tests and provides an API to retrieve them. See documentation at reporting/resources/README.md
```
Comment (sideshowbarker) [inline docs/writing-tests/interacting-features.md:25]:
```suggestion
There are several useful utilities in the `/common/` directory

### Cookies

Features which need to test their interaction with cookies can use the scripts in `cookies/resources` to control which cookies are set on a given request.

### Permissions Policy

Features which integrate with Permissions Policy can make use of the `permissions-policy.js` support library to generate a set of tests for that integration.

### Reporting

Testing integration with the Reporting API can be done with the help of the common report collector. This service will collect reports sent from tests and provides an API to retrieve them. See documentation at `reporting/resources/README.md`.
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by clelland** [comment]
Thanks for the quick review (and edits, and merge)!

**Pull request merged**